### PR TITLE
mrc-2059 Endpoints for async calibrate

### PR DESCRIPTION
**Pull request by EmmaLRussell**
## Description

Add backend endpoints for async calibration, for submit, status and result. 

## Type of version change
_Delete as appropriate_

None

## Checklist

- [x] I have incremented version number, or version needs no increment
- [x] The build passed successfully, or failed because of ADR tests

### CODE REVIEW

File: src/config/detekt/detekt.yml [hunk 1]
```diff
@@ -15,6 +15,7 @@ complexity:
     ignoreDeprecated: true
     ignorePrivate: true
     ignoreOverridden: false
+    excludes: ['**/HintrAPIClient.kt']
```
Comment (EmmaLRussell) [inline src/config/detekt/detekt.yml:18]: This linter exclusion can be temporary since we'll be able to remove the old `calibrate` method eventually

**Review by Nicolas-Dolan — APPROVED** [review]
When I ran this locally I got a few error messages coming the frontend, but I don't think they are directly related to this branch. The calibrate step itself ran smoothly though. Admittedly, my ability to review the code is limited without working through it to develop a full understanding of the code myself but I see that you have the three endpoints in place and it follows the model run pattern.

image

**Comment by EmmaLRussell** [comment]
Well spotted, I'll have a look into those. 

> When I ran this locally I got a few error messages coming the frontend, but I don't think they are directly related to this branch. The calibrate step itself ran smoothly though. Admittedly, my ability to review the code is limited without working through it to develop a full understanding of the code myself but I see that you have the three endpoints in place and it follows the model run pattern.

**Comment by EmmaLRussell** [comment]
OK, I think this must be because the backend is now returning an empty response from the initial run. I think the best thing will be to sort this in the next branch, which regenerates the front end types. So, I'll merge this since you've approved it, and do that!

> Well spotted, I'll have a look into those.
> 
> > When I ran this locally I got a few error messages coming the frontend, but I don't think they are directly related to this branch. The calibrate step itself ran smoothly though. Admittedly, my ability to review the code is limited without working through it to develop a full understanding of the code myself but I see that you have the three endpoints in place and it follows the model run pattern.

**Comment by EmmaLRussell** [comment]
Ugh, damnit, merged to master

**Pull request merged**